Add codecov.io into make tests and Travis

### General information

Add codecov.io integration into .travis.yml so that code coverage results are computed on each PR in travis and team would have additional check on PR to see whether a particular PR is not missing unit tests.

Additionally add test coverage generation into `make test` target - just adding `go test -cover` might be enough.

  * Minishift version: N/A
  * OS: N/A
  * Hypervisor: N/A

Some links:
* https://gist.github.com/jonfriesen/1c64bebdd2b30ba1b2465c259f070459
* https://github.com/codecov/example-go
* https://blog.golang.org/cover

### Steps to reproduce

  1. `make test`
  
### Expected

Code coverage results are generated.

### Actual

Code coverage results are not generated.

### Logs

Tried but I was not able to have a clean run of `make test` on macOS:

```
# github.com/minishift/minishift/pkg/util/os/atexit
pkg/util/os/atexit/atexit.go:77: Exitf call has arguments but no formatting directives
?   	github.com/minishift/minishift	[no test files]
?   	github.com/minishift/minishift/cmd/minishift	[no test files]
=== RUN   TestDisplayConsoleInMachineReadable
--- PASS: TestDisplayConsoleInMachineReadable (0.00s)
=== RUN   Test_clear_cache_user_confirms
--- PASS: Test_clear_cache_user_confirms (0.00s)
=== RUN   Test_clear_cache_user_aborts
--- PASS: Test_clear_cache_user_aborts (0.00s)
=== RUN   Test_clear_cache_forced
--- PASS: Test_clear_cache_forced (0.00s)
=== RUN   Test_delete_succeeds_for_non_existing_vm
--- PASS: Test_delete_succeeds_for_non_existing_vm (0.00s)
=== RUN   Test_unix_oc_path
--- PASS: Test_unix_oc_path (0.00s)
=== RUN   TestPreRunDirectories
--- PASS: TestPreRunDirectories (0.00s)
=== RUN   TestViperConfig
--- PASS: TestViperConfig (0.00s)
=== RUN   TestViperAndFlags
--- PASS: TestViperAndFlags (0.00s)
    viper.go:81: TestConfig: Unsetting MINISHIFT_GITHUB_API_TOKEN=c1a1501eed615c1a7eeb9e5aa98dd697512a5e56 for unit test!
    viper.go:87: TestConfig: Finished test, Resetting Env MINISHIFT_GITHUB_API_TOKEN=c1a1501eed615c1a7eeb9e5aa98dd697512a5e56
=== RUN   TestInitializeProfile
--- PASS: TestInitializeProfile (0.00s)
=== RUN   TestIsoUrl
--- PASS: TestIsoUrl (0.00s)
=== RUN   TestStartClusterUpNoFlags
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: TestStartClusterUpNoFlags (0.00s)
=== RUN   TestStartClusterUpWithFlag
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: TestStartClusterUpWithFlag (0.00s)
=== RUN   TestClusterUpWithProxyFlag
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: TestClusterUpWithProxyFlag (0.00s)
=== RUN   TestCheckMemorySize
--- PASS: TestCheckMemorySize (0.00s)
=== RUN   TestCheckDiskSize
--- PASS: TestCheckDiskSize (0.00s)
=== RUN   TestDetermineIsoUrl
--- PASS: TestDetermineIsoUrl (0.00s)
=== RUN   TestDetermineIsoUrlWithInvalidName
--- PASS: TestDetermineIsoUrlWithInvalidName (0.00s)
=== RUN   Test_getslice_withConfig
--- PASS: Test_getslice_withConfig (0.00s)
=== RUN   Test_getslice_withCommandLine
--- PASS: Test_getslice_withCommandLine (0.00s)
=== RUN   TestCreateUpdateMarker
--- PASS: TestCreateUpdateMarker (0.00s)
=== RUN   Test_version_output_has_correct_format
minishift v1.24.0+86e8eb56
--- PASS: Test_version_output_has_correct_format (0.00s)
PASS
coverage: 21.4% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd	(cached)	coverage: 21.4% of statements
=== RUN   Test_addon_name_must_be_specified_for_apply_command
--- PASS: Test_addon_name_must_be_specified_for_apply_command (0.00s)
=== RUN   Test_unknown_name_for_apply_command_returns_error
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: Test_unknown_name_for_apply_command_returns_error (0.00s)
=== RUN   Test_addon_name_must_be_specified_for_disable_command
--- PASS: Test_addon_name_must_be_specified_for_disable_command (0.00s)
=== RUN   Test_unknown_name_for_disable_command_returns_error
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: Test_unknown_name_for_disable_command_returns_error (0.00s)
=== RUN   Test_addon_name_must_be_specified_for_enable_command
--- PASS: Test_addon_name_must_be_specified_for_enable_command (0.00s)
=== RUN   Test_unknown_name_for_enable_command_returns_error
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: Test_unknown_name_for_enable_command_returns_error (0.00s)
=== RUN   Test_source_of_addon_must_be_specified
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: Test_source_of_addon_must_be_specified (0.00s)
=== RUN   Test_install_with_enable_flag_works
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: Test_install_with_enable_flag_works (0.00s)
=== RUN   Test_addon_name_must_be_specified_for_uninstall_command
--- PASS: Test_addon_name_must_be_specified_for_uninstall_command (0.00s)
=== RUN   Test_uninstall_with_enable_flag_works
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: Test_uninstall_with_enable_flag_works (0.00s)
PASS
coverage: 17.5% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd/addon	(cached)	coverage: 17.5% of statements
=== RUN   TestReadConfig
--- PASS: TestReadConfig (0.00s)
=== RUN   TestWriteConfig
--- PASS: TestWriteConfig (0.00s)
=== RUN   TestNoExplicitRouteSuffixDefaultsToNip
--- PASS: TestNoExplicitRouteSuffixDefaultsToNip (0.00s)
=== RUN   TestExplicitRouteSuffixGetApplied
--- PASS: TestExplicitRouteSuffixGetApplied (0.00s)
=== RUN   TestNotFound
--- PASS: TestNotFound (0.00s)
=== RUN   TestModifyData
No Minishift instance exists. New 'cpus' setting will be applied on next 'minishift start'
No Minishift instance exists. New 'cpus' setting will be applied on next 'minishift start'
--- PASS: TestModifyData (0.00s)
=== RUN   TestFindSettingNotFound
--- PASS: TestFindSettingNotFound (0.00s)
=== RUN   TestFindSetting
--- PASS: TestFindSetting (0.00s)
=== RUN   TestSetString
--- PASS: TestSetString (0.00s)
=== RUN   TestSetInt
--- PASS: TestSetInt (0.00s)
=== RUN   TestSetBool
--- PASS: TestSetBool (0.00s)
=== RUN   TestSetSlice
--- PASS: TestSetSlice (0.00s)
=== RUN   TestConfigView
--- PASS: TestConfigView (0.00s)
PASS
coverage: 60.7% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd/config	(cached)	coverage: 60.7% of statements
?   	github.com/minishift/minishift/cmd/minishift/cmd/dns	[no test files]
=== RUN   Test_host_folder_name_required
--- PASS: Test_host_folder_name_required (0.00s)
=== RUN   Test_host_folder_name_not_empty
--- PASS: Test_host_folder_name_not_empty (0.00s)
=== RUN   Test_source_required
--- PASS: Test_source_required (0.00s)
=== RUN   Test_target_required
--- PASS: Test_target_required (0.00s)
=== RUN   Test_username_required
--- PASS: Test_username_required (0.00s)
=== RUN   Test_password_required
--- PASS: Test_password_required (0.00s)
=== RUN   Test_domain_required
--- PASS: Test_domain_required (0.00s)
=== RUN   Test_unknown_host_folder_type_exits
--- PASS: Test_unknown_host_folder_type_exits (0.00s)
=== RUN   Test_get_options
--- PASS: Test_get_options (0.00s)
PASS
coverage: 27.8% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd/hostfolder	(cached)	coverage: 27.8% of statements
=== RUN   Test_no_images_to_export
--- PASS: Test_no_images_to_export (0.00s)
=== RUN   Test_no_images_to_import
Recovered from panic: runtime error: invalid memory address or nil pointer dereference
--- PASS: Test_no_images_to_import (0.00s)
=== RUN   Test_no_images_cached
--- PASS: Test_no_images_cached (0.00s)
=== RUN   Test_no_images_to_prune
--- PASS: Test_no_images_to_prune (0.00s)
=== RUN   Test_normalize_image_names
--- PASS: Test_normalize_image_names (0.00s)
PASS
coverage: 23.7% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd/image	(cached)	coverage: 23.7% of statements
=== RUN   Test_restart_command_needs_existing_vm
--- PASS: Test_restart_command_needs_existing_vm (0.00s)
=== RUN   Test_unknown_patch_target_aborts_command
--- PASS: Test_unknown_patch_target_aborts_command (0.00s)
=== RUN   Test_patch_cannot_be_empty
--- PASS: Test_patch_cannot_be_empty (0.00s)
=== RUN   Test_patch_needs_to_be_valid_JSON
--- PASS: Test_patch_needs_to_be_valid_JSON (0.00s)
=== RUN   Test_patch_commands_needs_existing_vm
--- PASS: Test_patch_commands_needs_existing_vm (0.00s)
=== RUN   TestVersionCommandNeedsExistingVm
--- PASS: TestVersionCommandNeedsExistingVm (0.00s)
=== RUN   Test_view_commands_needs_existing_vm
--- PASS: Test_view_commands_needs_existing_vm (0.00s)
=== RUN   Test_unknown_config_target_aborts_view_command
--- PASS: Test_unknown_config_target_aborts_view_command (0.00s)
PASS
coverage: 23.4% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd/openshift	(cached)	coverage: 23.4% of statements
=== RUN   Test_validate_args_exit_when_profilename_empty
--- PASS: Test_validate_args_exit_when_profilename_empty (0.00s)
=== RUN   Test_validate_args_exit_when_no_profilename
--- PASS: Test_validate_args_exit_when_no_profilename (0.00s)
=== RUN   Test_validate_args_exit_when_more_then_one_args
--- PASS: Test_validate_args_exit_when_more_then_one_args (0.00s)
PASS
coverage: 13.0% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd/profile	(cached)	coverage: 13.0% of statements
?   	github.com/minishift/minishift/cmd/minishift/cmd/registration	[no test files]
?   	github.com/minishift/minishift/cmd/minishift/cmd/services	[no test files]
=== RUN   TestReplaceEnv
--- PASS: TestReplaceEnv (0.00s)
=== RUN   TestIsValidProfileName
--- PASS: TestIsValidProfileName (0.00s)
PASS
coverage: 8.4% of statements
ok  	github.com/minishift/minishift/cmd/minishift/cmd/util	(cached)	coverage: 8.4% of statements
?   	github.com/minishift/minishift/cmd/minishift/state	[no test files]
?   	github.com/minishift/minishift/cmd/testing/cli	[no test files]
?   	github.com/minishift/minishift/pkg/minikube/assets	[no test files]
=== RUN   TestRemoteBoot2DockerURL
--- PASS: TestRemoteBoot2DockerURL (0.00s)
=== RUN   TestLocalBoot2DockerURL
--- PASS: TestLocalBoot2DockerURL (0.00s)
=== RUN   TestFollowLogsFlag
=== RUN   TestFollowLogsFlag/docker_logs_origin
=== RUN   TestFollowLogsFlag/docker_logs_origin_-f
--- PASS: TestFollowLogsFlag (1.06s)
    --- PASS: TestFollowLogsFlag/docker_logs_origin (0.11s)
    --- PASS: TestFollowLogsFlag/docker_logs_origin_-f (0.04s)
PASS
coverage: 8.7% of statements
ok  	github.com/minishift/minishift/pkg/minikube/cluster	(cached)	coverage: 8.7% of statements
=== RUN   TestDefaultMinishiftHome
--- PASS: TestDefaultMinishiftHome (0.00s)
=== RUN   TestMinishiftHomeViaEnvironment
--- PASS: TestMinishiftHomeViaEnvironment (0.00s)
PASS
coverage: 53.8% of statements
ok  	github.com/minishift/minishift/pkg/minikube/constants	(cached)	coverage: 53.8% of statements
?   	github.com/minishift/minishift/pkg/minikube/kubeconfig	[no test files]
=== RUN   TestRunNotDriver
--- PASS: TestRunNotDriver (0.00s)
=== RUN   TestRunDriver
--- PASS: TestRunDriver (0.00s)
PASS
coverage: 50.0% of statements
ok  	github.com/minishift/minishift/pkg/minikube/machine	(cached)	coverage: 50.0% of statements
=== RUN   TestNewSSHClient
--- PASS: TestNewSSHClient (0.28s)
=== RUN   TestNewSSHHost
--- PASS: TestNewSSHHost (0.00s)
=== RUN   TestNewSSHHostError
--- PASS: TestNewSSHHostError (0.00s)
=== RUN   TestTransfer
--- PASS: TestTransfer (0.65s)
PASS
coverage: 64.5% of statements
ok  	github.com/minishift/minishift/pkg/minikube/sshutil	(cached)	coverage: 64.5% of statements
?   	github.com/minishift/minishift/pkg/minikube/tests	[no test files]
=== RUN   Test_Metadata_creation_succeeds
--- PASS: Test_Metadata_creation_succeeds (0.00s)
=== RUN   Test_missing_name_returns_error
--- PASS: Test_missing_name_returns_error (0.00s)
=== RUN   Test_missing_description_returns_error
--- PASS: Test_missing_description_returns_error (0.00s)
=== RUN   Test_optional_metadata_is_retained
--- PASS: Test_optional_metadata_is_retained (0.00s)
=== RUN   Test_required_vars_meta_is_extracted
--- PASS: Test_required_vars_meta_is_extracted (0.00s)
=== RUN   Test_required_vars_empty_if_not_specified
--- PASS: Test_required_vars_empty_if_not_specified (0.00s)
=== RUN   Test_var_defaults_empty_if_not_specified
--- PASS: Test_var_defaults_empty_if_not_specified (0.00s)
=== RUN   Test_var_defaults_meta_is_extracted
--- PASS: Test_var_defaults_meta_is_extracted (0.00s)
=== RUN   Test_invalid_var_defaults_meta
--- PASS: Test_invalid_var_defaults_meta (0.00s)
=== RUN   Test_check_openshift_version_sematic
--- PASS: Test_check_openshift_version_sematic (0.00s)
=== RUN   Test_sorting_addons_by_priority
--- PASS: Test_sorting_addons_by_priority (0.00s)
=== RUN   Test_sorting_addons_by_state_and_name
--- PASS: Test_sorting_addons_by_state_and_name (0.00s)
=== RUN   Test_sorting_addons_by_state_priority_and_name
--- PASS: Test_sorting_addons_by_state_priority_and_name (0.00s)
PASS
coverage: 59.5% of statements
ok  	github.com/minishift/minishift/pkg/minishift/addon	(cached)	coverage: 59.5% of statements
=== RUN   Test_echo_command
--- PASS: Test_echo_command (0.00s)
=== RUN   Test_interpolate_cmd
--- PASS: Test_interpolate_cmd (0.00s)
=== RUN   Test_adding_multiple_interpolation_variables
--- PASS: Test_adding_multiple_interpolation_variables (0.00s)
=== RUN   Test_dollar_sign_in_replacement_value_prints_literally
--- PASS: Test_dollar_sign_in_replacement_value_prints_literally (0.00s)
=== RUN   Test_adding_and_removing_context_variables
--- PASS: Test_adding_and_removing_context_variables (0.00s)
=== RUN   Test_parsing_sleep_time_is_successful
--- PASS: Test_parsing_sleep_time_is_successful (0.00s)
=== RUN   Test_parsing_sleep_time_ignores_text_after_durationb
--- PASS: Test_parsing_sleep_time_ignores_text_after_durationb (0.00s)
=== RUN   Test_parsing_sleep_time_fails
--- PASS: Test_parsing_sleep_time_fails (0.00s)
PASS
coverage: 30.0% of statements
ok  	github.com/minishift/minishift/pkg/minishift/addon/command	(cached)	coverage: 30.0% of statements
?   	github.com/minishift/minishift/pkg/minishift/addon/config	[no test files]
=== RUN   Test_creating_addon_manager_for_non_existing_directory_returns_an_error
--- PASS: Test_creating_addon_manager_for_non_existing_directory_returns_an_error (0.00s)
=== RUN   Test_create_addon_manager
--- PASS: Test_create_addon_manager (0.00s)
=== RUN   Test_installing_addon_for_non_existing_directory_returns_an_error
--- PASS: Test_installing_addon_for_non_existing_directory_returns_an_error (0.00s)
=== RUN   Test_invalid_addons_get_skipped
Skipping addon 'addon2' in '/var/folders/_9/_hgvg43n7l5fhglg35tx66wc0000gn/T/minishift-test-addon-manager-365211030/addon2' due to parse error: Metadata does not contain a mandatory entry for 'Name'
--- PASS: Test_invalid_addons_get_skipped (0.00s)
=== RUN   TestAddVarDefaultsToContext
--- PASS: TestAddVarDefaultsToContext (0.00s)
=== RUN   TestVerifyValidRequiredVariablesInContext
--- PASS: TestVerifyValidRequiredVariablesInContext (0.00s)
=== RUN   TestVerifyMissingRequiredVariablesInContext
--- PASS: TestVerifyMissingRequiredVariablesInContext (0.00s)
=== RUN   TestApplyAddon
Skipping addon 'addon-dir-mismatch' in '/Users/kpiwko/go/src/github.com/minishift/minishift/test/testdata/testaddons/addon-dir-mismatch' due to parse error: Add-on directory name should match to addon name
-- Applying addon 'testaddon':
This testaddon is having variable TEST with foo value
--- PASS: TestApplyAddon (0.00s)
=== RUN   TestRemoveAddon
Skipping addon 'addon-dir-mismatch' in '/Users/kpiwko/go/src/github.com/minishift/minishift/test/testdata/testaddons/addon-dir-mismatch' due to parse error: Add-on directory name should match to addon name
-- Removing addon 'testaddon':
Removing testaddon with variable TEST of foo value
--- PASS: TestRemoveAddon (0.00s)
=== RUN   TestRemoveMetaAddon
Skipping addon 'addon-dir-mismatch' in '/Users/kpiwko/go/src/github.com/minishift/minishift/test/testdata/testaddons/addon-dir-mismatch' due to parse error: Add-on directory name should match to addon name
-- Removing addon 'metaaddon':
Removing metaaddon without depending on addon default variables
--- PASS: TestRemoveMetaAddon (0.00s)
=== RUN   TestMetadataAddon
Skipping addon 'addon-dir-mismatch' in '/Users/kpiwko/go/src/github.com/minishift/minishift/test/testdata/testaddons/addon-dir-mismatch' due to parse error: Add-on directory name should match to addon name
-- Applying addon 'metaaddon':--- PASS: TestMetadataAddon (0.00s)
=== RUN   TestApplyInvalidAddon
Skipping addon 'addon-dir-mismatch' in '/Users/kpiwko/go/src/github.com/minishift/minishift/test/testdata/testaddons/addon-dir-mismatch' due to parse error: Add-on directory name should match to addon name
-- Applying addon 'invalidaddon':--- PASS: TestApplyInvalidAddon (0.00s)
=== RUN   TestRemoveInvalidAddon
Skipping addon 'addon-dir-mismatch' in '/Users/kpiwko/go/src/github.com/minishift/minishift/test/testdata/testaddons/addon-dir-mismatch' due to parse error: Add-on directory name should match to addon name
-- Removing addon 'invalidaddon':--- PASS: TestRemoveInvalidAddon (0.00s)
PASS
coverage: 45.8% of statements
ok  	github.com/minishift/minishift/pkg/minishift/addon/manager	(cached)	coverage: 45.8% of statements
=== RUN   Test_successful_parsing_of_addon_dir_without_remove_addon_file
--- PASS: Test_successful_parsing_of_addon_dir_without_remove_addon_file (0.00s)
=== RUN   Test_successful_parsing_of_addon_dir_with_remove_addon_file
--- PASS: Test_successful_parsing_of_addon_dir_with_remove_addon_file (0.00s)
=== RUN   Test_empty_lines_and_comments_in_content_are_ignored
--- PASS: Test_empty_lines_and_comments_in_content_are_ignored (0.00s)
=== RUN   Test_multiple_lines_in_description_with_tab_and_space
--- PASS: Test_multiple_lines_in_description_with_tab_and_space (0.00s)
=== RUN   Test_multiple_lines_in_description_with_comment
--- PASS: Test_multiple_lines_in_description_with_comment (0.00s)
=== RUN   Test_addon_with_unknown_command_returns_error
--- PASS: Test_addon_with_unknown_command_returns_error (0.00s)
=== RUN   Test_non_existend_addon_directory_creates_error
--- PASS: Test_non_existend_addon_directory_creates_error (0.00s)
=== RUN   Test_successful_parsing_of_content
--- PASS: Test_successful_parsing_of_content (0.00s)
=== RUN   Test_all_command_types_recognized
--- PASS: Test_all_command_types_recognized (0.00s)
=== RUN   Test_metadata_creation_requires_name_attribute
--- PASS: Test_metadata_creation_requires_name_attribute (0.00s)
=== RUN   Test_metadata_creation_requires_description_attribute
--- PASS: Test_metadata_creation_requires_description_attribute (0.00s)
=== RUN   Test_at_least_one_addon_file_required
--- PASS: Test_at_least_one_addon_file_required (0.00s)
=== RUN   Test_multiple_addon_definitions_create_error
--- PASS: Test_multiple_addon_definitions_create_error (0.00s)
=== RUN   TestInvalidAddonDirAndName
--- PASS: TestInvalidAddonDirAndName (0.00s)
PASS
coverage: 89.5% of statements
ok  	github.com/minishift/minishift/pkg/minishift/addon/parser	(cached)	coverage: 89.5% of statements
=== RUN   TestIsCached
--- PASS: TestIsCached (0.00s)
=== RUN   TestCacheOc
OK--- PASS: TestCacheOc (1.75s)
PASS
coverage: 53.8% of statements
ok  	github.com/minishift/minishift/pkg/minishift/cache	(cached)	coverage: 53.8% of statements
?   	github.com/minishift/minishift/pkg/minishift/cluster	[no test files]
=== RUN   Test_invalid_addon_variable_leads_to_error_in_context_creation
--- PASS: Test_invalid_addon_variable_leads_to_error_in_context_creation (0.00s)
=== RUN   Test_addon_variable_can_be_interpolated
--- PASS: Test_addon_variable_can_be_interpolated (0.00s)
=== RUN   Test_nil_can_be_passed_to_create_context
--- PASS: Test_nil_can_be_passed_to_create_context (0.00s)
=== RUN   Test_addon_variable_can_be_interpolated_from_environment
--- PASS: Test_addon_variable_can_be_interpolated_from_environment (0.00s)
PASS
coverage: 19.4% of statements
ok  	github.com/minishift/minishift/pkg/minishift/clusterup	(cached)	coverage: 19.4% of statements
=== RUN   TestNewInstanceConfig
--- PASS: TestNewInstanceConfig (0.00s)
=== RUN   TestConfigOnFileExists
--- PASS: TestConfigOnFileExists (0.00s)
=== RUN   TestWrite
--- PASS: TestWrite (0.00s)
=== RUN   TestDelete
--- PASS: TestDelete (0.00s)
=== RUN   TestDriver
--- PASS: TestDriver (0.00s)
=== RUN   TestValidCIDR
--- PASS: TestValidCIDR (0.00s)
=== RUN   TestValidURL
--- PASS: TestValidURL (0.00s)
=== RUN   TestValidProxyURL
--- PASS: TestValidProxyURL (0.00s)
=== RUN   TestValidIPv4Address
--- PASS: TestValidIPv4Address (0.00s)
=== RUN   TestValidNetmask
--- PASS: TestValidNetmask (0.00s)
=== RUN   TestValidPort
--- PASS: TestValidPort (0.00s)
PASS
coverage: 33.5% of statements
ok  	github.com/minishift/minishift/pkg/minishift/config	(cached)	coverage: 33.5% of statements
?   	github.com/minishift/minishift/pkg/minishift/constants	[no test files]
?   	github.com/minishift/minishift/pkg/minishift/docker	[no test files]
=== RUN   Test_Are_Images_Cached
--- PASS: Test_Are_Images_Cached (0.00s)
=== RUN   Test_Get_Docker_Settings
--- PASS: Test_Get_Docker_Settings (0.00s)
PASS
coverage: 14.1% of statements
ok  	github.com/minishift/minishift/pkg/minishift/docker/image	(cached)	coverage: 14.1% of statements
=== RUN   Test_type_string
--- PASS: Test_type_string (0.00s)
PASS
coverage: 1.2% of statements
ok  	github.com/minishift/minishift/pkg/minishift/hostfolder	(cached)	coverage: 1.2% of statements
=== RUN   TestHostFolderConfig
--- PASS: TestHostFolderConfig (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/minishift/minishift/pkg/minishift/hostfolder/config	(cached)	coverage: 100.0% of statements
=== RUN   TestFillNetworkSettingsScript
--- PASS: TestFillNetworkSettingsScript (0.00s)
PASS
coverage: 4.8% of statements
ok  	github.com/minishift/minishift/pkg/minishift/network	(cached)	coverage: 4.8% of statements
?   	github.com/minishift/minishift/pkg/minishift/network/dns	[no test files]
?   	github.com/minishift/minishift/pkg/minishift/network/proxy	[no test files]
=== RUN   Test_invalid_oc_path_returns_error
--- PASS: Test_invalid_oc_path_returns_error (0.00s)
=== RUN   Test_invalid_kube_path_returns_error
--- PASS: Test_invalid_kube_path_returns_error (0.00s)
=== RUN   Test_quotes_in_commands_get_properly_parsed
--- PASS: Test_quotes_in_commands_get_properly_parsed (0.00s)
=== RUN   TestParseOcHelpCommand
--- PASS: TestParseOcHelpCommand (0.00s)
=== RUN   TestFlagExist
--- PASS: TestFlagExist (0.00s)
PASS
coverage: 31.6% of statements
ok  	github.com/minishift/minishift/pkg/minishift/oc	(cached)	coverage: 31.6% of statements
=== RUN   Test_getServiceSpecs_with_multiple_nodeports_and_routes
--- PASS: Test_getServiceSpecs_with_multiple_nodeports_and_routes (0.00s)
PASS
coverage: 34.2% of statements
ok  	github.com/minishift/minishift/pkg/minishift/openshift	(cached)	coverage: 34.2% of statements
=== RUN   TestPrintUpstreamVersions
--- PASS: TestPrintUpstreamVersions (1.67s)
=== RUN   TestIsGreaterOrEqualToBaseVersion
--- PASS: TestIsGreaterOrEqualToBaseVersion (0.00s)
=== RUN   TestIsPrerelease
--- PASS: TestIsPrerelease (0.00s)
=== RUN   TestOpenShiftTagsByAscending
--- PASS: TestOpenShiftTagsByAscending (0.00s)
ok  	github.com/minishift/minishift/pkg/minishift/openshift/version	(cached)
=== RUN   TestSetingActiveProfile
--- PASS: TestSetingActiveProfile (0.00s)
=== RUN   TestSetingGetProfileList
--- PASS: TestSetingGetProfileList (0.00s)
PASS
coverage: 61.8% of statements
ok  	github.com/minishift/minishift/pkg/minishift/profile	(cached)	coverage: 61.8% of statements
=== RUN   TestBuildRootLogLevel

   Setting hostname ... 
FAIL
--- PASS: TestBuildRootLogLevel (0.00s)
=== RUN   TestBuildRootProvisionerGenerateDockerOptions
--- PASS: TestBuildRootProvisionerGenerateDockerOptions (0.00s)
=== RUN   TestMinishiftProvisionerSelected
Waiting for SSH to be available...
Detecting the provisioner...
--- PASS: TestMinishiftProvisionerSelected (0.16s)
=== RUN   TestDefaultCentOSProvisionerSelected
Waiting for SSH to be available...
Detecting the provisioner...
--- PASS: TestDefaultCentOSProvisionerSelected (0.84s)
=== RUN   TestDefaultRHELOSProvisionerSelected
Waiting for SSH to be available...
Detecting the provisioner...
--- PASS: TestDefaultRHELOSProvisionerSelected (0.27s)
=== RUN   TestDefaultUbuntuProvisionerSelected
Waiting for SSH to be available...
Detecting the provisioner...
Waiting for SSH to be available...
Detecting the provisioner...
--- PASS: TestDefaultUbuntuProvisionerSelected (0.86s)
=== RUN   TestMinishiftLogLevel

   Setting hostname ... 
FAIL
--- PASS: TestMinishiftLogLevel (0.00s)
=== RUN   TestMinishiftDefaultStorageDriver

   Setting hostname ... 
FAIL
--- PASS: TestMinishiftDefaultStorageDriver (0.00s)
=== RUN   TestRhelImage
--- PASS: TestRhelImage (0.00s)
=== RUN   TestCentOSImage
--- PASS: TestCentOSImage (0.00s)
=== RUN   TestMinishiftProvisionerGenerateDockerOptionsForRHEL
--- PASS: TestMinishiftProvisionerGenerateDockerOptionsForRHEL (0.00s)
=== RUN   TestMinishiftProvisionerGenerateDockerOptionsForCentOS
--- PASS: TestMinishiftProvisionerGenerateDockerOptionsForCentOS (0.00s)
PASS
coverage: 47.5% of statements
ok  	github.com/minishift/minishift/pkg/minishift/provisioner	(cached)	coverage: 47.5% of statements
=== RUN   TestRedHatRegistratorCompatibleWithDistribution
--- PASS: TestRedHatRegistratorCompatibleWithDistribution (0.89s)
=== RUN   TestRedHatRegistratorNotCompatibleWithDistribution
--- PASS: TestRedHatRegistratorNotCompatibleWithDistribution (0.10s)
=== RUN   TestRedHatRegistratorRegister
   Registration in progress . OK [2s]
--- PASS: TestRedHatRegistratorRegister (2.20s)
=== RUN   TestRedHatRegistratorUnregister
--- PASS: TestRedHatRegistratorUnregister (0.76s)
PASS
coverage: 32.3% of statements
ok  	github.com/minishift/minishift/pkg/minishift/registration	(cached)	coverage: 32.3% of statements
?   	github.com/minishift/minishift/pkg/minishift/remotehost	[no test files]
?   	github.com/minishift/minishift/pkg/minishift/shell/powershell	[no test files]
?   	github.com/minishift/minishift/pkg/minishift/systemd	[no test files]
?   	github.com/minishift/minishift/pkg/minishift/systemd/action	[no test files]
?   	github.com/minishift/minishift/pkg/minishift/tls	[no test files]
=== RUN   TestDownloadAndVerifyArchive
Downloading https://github.com/minishift/minishift/releases/download/v1.0.0/minishift-1.0.0-linux-amd64.tgz
Downloading https://github.com/minishift/minishift/releases/download/v1.0.0/minishift-1.0.0-windows-amd64.zip
Downloading https://github.com/minishift/minishift/releases/download/v1.2.0/minishift-1.2.0-linux-amd64.tgz
Downloading https://github.com/minishift/minishift/releases/download/v1.2.0/minishift-1.2.0-windows-amd64.zip
Downloading https://github.com/minishift/minishift/releases/download/v1.2.1/minishift-1.2.1-linux-amd64.tgz
Downloading https://github.com/minishift/minishift/releases/download/v1.2.1/minishift-1.2.1-windows-amd64.zip
--- PASS: TestDownloadAndVerifyArchive (0.01s)
=== RUN   TestExtractBinaryforOlderVersionFormat
--- PASS: TestExtractBinaryforOlderVersionFormat (0.00s)
=== RUN   TestExtractBinaryforNewerVersionFormat
--- PASS: TestExtractBinaryforNewerVersionFormat (0.00s)
=== RUN   TestExtractBinaryWithoutHavingMiniShift
--- PASS: TestExtractBinaryWithoutHavingMiniShift (0.00s)
PASS
coverage: 36.0% of statements
ok  	github.com/minishift/minishift/pkg/minishift/update	(cached)	coverage: 36.0% of statements
=== RUN   TestGetIsoPath
--- PASS: TestGetIsoPath (0.00s)
=== RUN   Test_getMinikubeIsoVersion
--- PASS: Test_getMinikubeIsoVersion (0.00s)
PASS
coverage: 58.3% of statements
ok  	github.com/minishift/minishift/pkg/minishift/util	(cached)	coverage: 58.3% of statements
=== RUN   Test_tee_captures_stdout_and_stderr
--- PASS: Test_tee_captures_stdout_and_stderr (0.00s)
PASS
coverage: 37.6% of statements
ok  	github.com/minishift/minishift/pkg/testing	(cached)	coverage: 37.6% of statements
=== RUN   TestEncryption
--- PASS: TestEncryption (0.00s)
=== RUN   Test_default_no_proxy_list
--- PASS: Test_default_no_proxy_list (0.00s)
=== RUN   Test_proxy_config_as_slice
--- PASS: Test_proxy_config_as_slice (0.00s)
=== RUN   Test_set_to_environment
--- PASS: Test_set_to_environment (0.00s)
=== RUN   Test_add_no_proxy
--- PASS: Test_add_no_proxy (0.00s)
=== RUN   Test_validate_proxy_url
--- PASS: Test_validate_proxy_url (0.00s)
=== RUN   Test_http_proxy_from_env
--- PASS: Test_http_proxy_from_env (0.00s)
=== RUN   Test_http_proxy_from_env_lowercase_precedence
--- PASS: Test_http_proxy_from_env_lowercase_precedence (0.00s)
=== RUN   Test_parse_special_character_uri
--- PASS: Test_parse_special_character_uri (0.00s)
=== RUN   TestErrorGenerator
--- PASS: TestErrorGenerator (0.00s)
=== RUN   TestRetry
--- PASS: TestRetry (0.00s)
=== RUN   TestMultiError
--- PASS: TestMultiError (0.00s)
=== RUN   TestFriendlyDuration
--- PASS: TestFriendlyDuration (0.00s)
=== RUN   Test_command_executes_successfully_with_absolute_path
--- PASS: Test_command_executes_successfully_with_absolute_path (0.03s)
=== RUN   Test_command_executes_successfully_with_command_lookup
--- PASS: Test_command_executes_successfully_with_command_lookup (0.04s)
=== RUN   Test_command_executes_unsuccessfully_with_command_lookup
--- PASS: Test_command_executes_unsuccessfully_with_command_lookup (0.00s)
=== RUN   TestWritableDirectory
--- PASS: TestWritableDirectory (0.00s)
=== RUN   TestNonWritableDirectory
--- PASS: TestNonWritableDirectory (0.01s)
PASS
coverage: 63.6% of statements
ok  	github.com/minishift/minishift/pkg/util	(cached)	coverage: 63.6% of statements
?   	github.com/minishift/minishift/pkg/util/archive	[no test files]
=== RUN   TestReturnsAnArrayOfStrings
--- PASS: TestReturnsAnArrayOfStrings (0.00s)
=== RUN   TestRemovesOuterSingleQuotes
--- PASS: TestRemovesOuterSingleQuotes (0.00s)
=== RUN   TestRemovesOuterDoubleQuotes
--- PASS: TestRemovesOuterDoubleQuotes (0.00s)
=== RUN   TestProperlyHandlesWhitespaceWithinSingleQuotes
--- PASS: TestProperlyHandlesWhitespaceWithinSingleQuotes (0.00s)
=== RUN   TestProperlyHandlesWhitespaceWithinDoubleQuotes
--- PASS: TestProperlyHandlesWhitespaceWithinDoubleQuotes (0.00s)
=== RUN   TestProperlyHandlesEscapedDoubleQuotesWithinDoubleQuotes
--- PASS: TestProperlyHandlesEscapedDoubleQuotesWithinDoubleQuotes (0.00s)
=== RUN   TestProperlyHandlesNestedSingleQuotesWithinDoubleQuotes
--- PASS: TestProperlyHandlesNestedSingleQuotesWithinDoubleQuotes (0.00s)
=== RUN   TestProperlyHandlesNestedSingleQuotesInNestedDoubleQuotes
--- PASS: TestProperlyHandlesNestedSingleQuotesInNestedDoubleQuotes (0.00s)
PASS
coverage: 100.0% of statements
ok  	github.com/minishift/minishift/pkg/util/cmd	(cached)	coverage: 100.0% of statements
=== RUN   Test_non_existent_directory_returns_false
--- PASS: Test_non_existent_directory_returns_false (0.00s)
=== RUN   Test_existent_directory_returns_true
--- PASS: Test_existent_directory_returns_true (0.00s)
=== RUN   Test_testDir_is_directory
--- PASS: Test_testDir_is_directory (0.00s)
=== RUN   Test_non_existing_file_is_not_a_directory
--- PASS: Test_non_existing_file_is_not_a_directory (0.00s)
=== RUN   Test_file_is_not_a_directory
--- PASS: Test_file_is_not_a_directory (0.00s)
=== RUN   Test_non_existing_directory
--- PASS: Test_non_existing_directory (0.00s)
=== RUN   Test_existing_empty_directory
--- PASS: Test_existing_empty_directory (0.00s)
=== RUN   Test_existing_nonempty_directory
--- PASS: Test_existing_nonempty_directory (0.00s)
PASS
coverage: 20.0% of statements
ok  	github.com/minishift/minishift/pkg/util/filehelper	(cached)	coverage: 20.0% of statements
=== RUN   TestGetAssetIdAndFilename
--- PASS: TestGetAssetIdAndFilename (1.29s)
=== RUN   TestDownloadOc
-- Downloading OpenShift binary 'oc' version 'v1.3.1'
-- Downloading OpenShift v1.3.1 checksums ... OK-- Downloading OpenShift binary 'oc' version 'v1.3.1'
-- Downloading OpenShift v1.3.1 checksums ... OK-- Downloading OpenShift binary 'oc' version 'v1.3.1'
-- Downloading OpenShift v1.3.1 checksums ... OK--- PASS: TestDownloadOc (4.45s)
=== RUN   TestInvalidVersion
--- PASS: TestInvalidVersion (0.16s)
=== RUN   TestInvalidBinaryFormat
--- PASS: TestInvalidBinaryFormat (0.20s)
=== RUN   Test_Download_Oc_1_4_1
-- Downloading OpenShift binary 'oc' version 'v1.4.1'
-- Downloading OpenShift v1.4.1 checksums ... OK--- PASS: Test_Download_Oc_1_4_1 (1.61s)
PASS
coverage: 76.2% of statements
ok  	github.com/minishift/minishift/pkg/util/github	(cached)	coverage: 76.2% of statements
?   	github.com/minishift/minishift/pkg/util/os	[no test files]
=== RUN   TestUnknownShell
--- PASS: TestUnknownShell (0.00s)
=== RUN   TestKnownShell
--- PASS: TestKnownShell (0.00s)
=== RUN   TestNoProxyFromEnvWhenUsedLowerCase
--- PASS: TestNoProxyFromEnvWhenUsedLowerCase (0.00s)
=== RUN   TestNoProxyFromEnvWhenUsedUpperCase
--- PASS: TestNoProxyFromEnvWhenUsedUpperCase (0.00s)
=== RUN   TestGenerateUsageHint
--- PASS: TestGenerateUsageHint (0.00s)
PASS
coverage: 37.9% of statements
ok  	github.com/minishift/minishift/pkg/util/shell	(cached)	coverage: 37.9% of statements
=== RUN   TestContains
--- PASS: TestContains (0.00s)
=== RUN   TestRemove
--- PASS: TestRemove (0.00s)
=== RUN   TestHasMatcher
--- PASS: TestHasMatcher (0.00s)
=== RUN   TestGetMatcher
--- PASS: TestGetMatcher (0.00s)
=== RUN   TestEscapeSingleQuote
--- PASS: TestEscapeSingleQuote (0.00s)
=== RUN   TestSplitAndTrim
--- PASS: TestSplitAndTrim (0.00s)
PASS
coverage: 82.4% of statements
ok  	github.com/minishift/minishift/pkg/util/strings	(cached)	coverage: 82.4% of statements
make: *** [test] Error 2
```

And I guess I would need to have admin/write access to the repo to add it it codecov.io.

--- a/.codecov.yml
+++ b/.codecov.yml
@@ -1,0 +1,7 @@
+comment:
+  layout: "reach, diff, flags, files"
+  behavior: new
+  require_changes: false  # if true: only post the comment if coverage changes
+  require_base: no        # [yes :: must have a base report to post]
+  require_head: no       # [yes :: must have a head report to post]
+  branches: null

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,8 +18,13 @@ install:
 - chmod +x /tmp/godep/dep
 - export PATH=$PATH:/tmp/godep/
 
-script:
-- make prerelease
+script: |
+  make prerelease
+  make coverage
+
+after_success: |
+  cd out
+  bash <(curl -s https://codecov.io/bash)
 
 notifications:
   webhooks: http://minibot.19cf262c.svc.dockerapp.io:9009/hubot/travis-ci

--- a/Makefile
+++ b/Makefile
@@ -221,6 +221,11 @@ clean_bindata:
 test: $(ADDON_ASSET_FILE)  ## Run unit tests
 	@go test -v -tags "$(BUILD_TAGS_SYSTEMTRAY)" -ldflags="$(VERSION_VARIABLES)" $(shell $(PACKAGES))
 
+.PHONY: coverage
+coverage: $(ADDON_ASSET_FILE)
+	rm -f out/coverage.txt
+	@go test -v -tags "$(BUILD_TAGS_SYSTEMTRAY)" -ldflags="$(VERSION_VARIABLES)" -coverprofile=out/coverage.txt -covermode=atomic $(shell $(PACKAGES))
+
 .PHONY: integration ## Run integration tests (quick and minimal)
 integration: GODOG_OPTS = --tags=quick\&\&~disabled
 integration: $(MINISHIFT_BINARY)

--- a/docs/source/contributing/developing.adoc
+++ b/docs/source/contributing/developing.adoc
@@ -185,6 +185,14 @@ For more {project} commands and flags, see the xref:../command-ref/minishift.ado
 === Unit Tests
 
 Unit tests run on *Travis* before the code is merged.
+Unit test coverage is then reported to *Codecov* which provides coverage information on the pull requests.
+Locally you can generate a HTML code coverage report by following steps:
+
+----
+$ make coverage
+$ go tool cover -html=out/coverage.txt -o out/coverage.html
+----
+
 To run tests during the development cycle:
 
 ----

